discussions/secure_web_api.md

```diff
@@ -0,0 +1,137 @@
+  programs will run, their execution order, and the configuration arguments for each eBPF program.
+- A database and local KV store that stores the configuration data. 
+- A datastore that stores the eBPF program artifact (I.e., byte code and native code).
+
+When users want to deploy an eBPF program, they can call the L3AFD API with appropriate parameters. This request would
+generate a new config (KV pair). Once L3AFD reads this new config, it orchestrates eBPF programs on the Linux host
+as per the defined parameters. If the user gives a set of eBPF programs, then L3AFD can orchestrate all of them in the
+sequence that the user wanted (aka chaining).
+
+## Need for secure web APIs
+
+L3AFD uses Go HTTPS client to download the configured eBPF packages from a datastore (package repository).
+However, this client does not support TLS. In the open-source world, users will presumably expect to be able
+to use TLS in this situation. This is also an early step toward using a secure eBPF Package Repository
+(https://github.com/l3af-project/l3afd/issues/2).
+
+L3AFD has a gRPC and HTTP API that can be used to configure the eBPF programs. However, this API only supports HTTP.
+This works for a use case where the L3AFD API is called only from the localhost, but this is probably unintuitive.
+In the future, we would like to come up with service that can call the L3AFD APIs remotely. For this, we could leverage
+mTLS for a mutually secure connection between the client and server. Also, new open-source adopters of L3AF will presumably
+want to avoid calling the L3AFD API locally on each node (https://github.com/l3af-project/l3afd/issues/4).
+
+## Types of certificates are supported
+
+The TLS protocol aims primarily to provide cryptography, including privacy (confidentiality), integrity, and
+authenticity through certificates, between L3AFD and client's communication. These certificates can be issued by a
+third-party trusted authority (i.e., IdenTrust, DigiCert, Sectigo, etc), and we can create self-signed certificates
+using tools like ```openssl```.
+
+This completely depends on the users, whether to use Trusted CA certificates or Self signed certificates. L3AF will not 
+provide any certificates.
+
+## L3AF deployment scenarios
+
+L3AF could be running in two scenarios, users can use L3AF in secure enterprise private networks and in public network.
+In case of private network, L3AFD and clients will be communicating with each other over a network that is normally
+protected by vpn or PCI (Payment Card Information), and hence it may not be essential to enable mTLS in this case.
+
+However, in case of public network, clients will be communicating with L3AFD over insecure networks. L3AF wants to
+secure its endpoints using industry's best standard available solutions. This can be configured from l3afd’s config file,
+and by default, mTLS will be disabled.
```

Comment:
```suggestion
secure its endpoints using industry's best standard available solutions. This can be configured from l3afd’s config file.
```
I don't think security should be disabled by default, so better to omit that phrase